table/full_filter_block.cc

```diff
@@ -153,4 +157,50 @@ bool FullFilterBlockReader::MayMatch(const Slice& entry) {
 size_t FullFilterBlockReader::ApproximateMemoryUsage() const {
   return contents_.size();
 }
+
+bool FullFilterBlockReader::IsFilterCompatible(
+    const ReadOptions& read_options, const Slice& user_key,
+    const Comparator* comparator) {
+  // TODO (Zhongyi): consider removing table_prefix_extractor
+  // Try to reuse the bloom filter in the SST table if prefix_extractor in
+  // mutable_cf_options has changed. If range [user_key, upper_bound) all
+  // share the same prefix then we may still be able to use the bloom filter.
+  if (read_options.iterate_upper_bound != nullptr && prefix_extractor_) {
+    if (!prefix_extractor_->InDomain(user_key) ||
+        !prefix_extractor_->InDomain(
+            *read_options.iterate_upper_bound)) {
+      return false;
+    }
+    Slice user_key_xform = prefix_extractor_->Transform(user_key);
+    Slice upper_bound_xform = prefix_extractor_->Transform(
+        *read_options.iterate_upper_bound);
+    // first check if user_key and upper_bound all share the same prefix
+    if (user_key_xform.compare(upper_bound_xform) != 0) {
```

Comment:
It's not a big deal but to be strict, we should call `Comparator::IsEqual(user_key_xform, upper_bound_xform)` here.